torch/_subclasses/fake_impls.py

```diff
@@ -147,6 +147,11 @@ def dispatch_to_op_implementations_dict(fake_mode, func, *args, **kwargs):
     return op_implementations_dict[func](fake_mode, func, *args, **kwargs)
 
 
+@register_op_impl(torch.ops.aten.sym_data_ptr.default)
+def _(fake_mode, func, x):
+    return fake_mode.shape_env.create_unbacked_symint()
```

Comment:
For numel 0 tensors, you could just return 0. and otherwise, you could add a lower bound >= 8 (if not higher, but 8 at least traces through 0/1 checks)..